plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -302,6 +302,7 @@ protected void dropTable(ConnectorSession session, RemoteTableName remoteTableNa
         String dropTableSql = "DROP TABLE " + quotedTable;
         try (Connection connection = connectionFactory.openConnection(session)) {
             if (temporaryTable) {
+                connection.setAutoCommit(false);
```

Comment:
what is the motivation behind the change? Should we commit explicitly then? Or was switching auto-commit back to true just before connection is closed for that?

Reply:
![image](https://github.com/trinodb/trino/assets/890111/ecd8dae9-d1d8-48b4-80e2-4eaf83eb8222)

Reply:
I see:

![image](https://github.com/trinodb/trino/assets/890111/95298f00-f8f3-4453-9ad8-7ae73f4c2d3e)

Reply:
Please add a comment below - or just call `commit()` - will be more readable I think

Reply:
The motivation is that, because autoCommit is true when we get the Connection, the lock is immediately released because that `LOCK TABLE` statement is auto-committed, so the Connection has to re-acquire the lock for the `DROP TABLE`, so we have the same situation as we did before. Turning off auto commit holds onto that lock until after the drop table. Oracle's docs say DDL always auto-commits, but I reset autoCommit to true after the drop just in case.

Reply:
Oh, weird - when i typed out my reply, i only had your first comment in this thread visible to me.

I think i want to keep it as `setAutoCommit(true)` as it properly handles both cases: 
![image](https://github.com/trinodb/trino/assets/2020261/8421071d-c80a-41ac-a001-4bea9de8b7f0)